src/source/StreamInfoProvider.c

```diff
@@ -49,31 +70,59 @@ STATUS freeStreamInfoProvider(PStreamInfo* ppStreamInfo)
     LEAVES();
     return retStatus;
 }
-
-// Creates H264 track info for video
-STATUS createH264VideoTrackInfo(PTrackInfo pTrackInfo)
+// Creates track info for video with given codec and sets the content type as per the codec
+STATUS createVideoTrackInfo(VIDEO_CODEC_ID videoCodecId, PCHAR contentType, PTrackInfo pTrackInfo)
 {
     STATUS retStatus = STATUS_SUCCESS;
 
     pTrackInfo->trackId = DEFAULT_VIDEO_TRACK_ID;
     pTrackInfo->codecPrivateData = NULL;
     pTrackInfo->codecPrivateDataSize = 0;
-    STRCPY(pTrackInfo->codecId, MKV_H264_AVC_CODEC_ID);
+    switch (videoCodecId) {
+        case VIDEO_CODEC_ID_H264:
+            STRCPY(pTrackInfo->codecId, MKV_H264_AVC_CODEC_ID);
+            STRCAT(contentType, MKV_H264_CONTENT_TYPE);
```

Comment:
I see this works because we calloc the streaminfo stuct so the first byte is all 0's, i.e. null char.